sklearn_extra/robust/robust_weighted_estimator.py

```diff
@@ -0,0 +1,486 @@
+
+        # Get actual loss function from its name.
+        loss = self._get_loss_function(loss_param)
+
+        # Weight initialization : do one non-robust epoch.
+        if loss_param in ["log", "hinge"]:
+            classes = np.unique(y)
+            if len(classes) > 2:
+                raise ValueError("y must be binary.")
+            # If in a classification task, precise the classes.
+            base_estimator.partial_fit(X, y, classes=classes)
+        else:
+            base_estimator.partial_fit(X, y)
+
+        # Initialization of final weights
+        final_weight = np.zeros(len(X))
+
+        # Optimization algorithm
+        for epoch in range(self.max_iter):
+
+            if epoch > self.burn_in and self.burn_in > 0:
+                # If not in the burn_in phase anymore, change the learning_rate
+                # calibration to the one edicted by self.base_estimator.
+                base_estimator.set_params(learning_rate=learning_rate)
+
+            if loss_param in ["log", "hinge"]:
+                # If in classification, use decision_function
+                pred = base_estimator.decision_function(X)
+            else:
+                pred = base_estimator.predict(X)
+
+            # Compute the loss of each sample
+            if self._estimator_type == "clusterer":
+                loss_values = loss(X, pred)
+            else:
+                loss_values = loss(y.flatten(), pred)
+
+            # Compute the weight associated with each losses.
+            # Samples whose loss is far from the mean loss (robust estimation)
+            # will have a small weight.
+            weights = self._get_weights(loss_values, random_state)
+
+            # Use the optimization algorithm of self.base_estimator for one
+            # epoch using the previously computed weights.
+            base_estimator.partial_fit(X, y, sample_weight=weights)
+
+            # Shuffle the data at each step.
+            if self._estimator_type == "clusterer":
+                X, weights = shuffle(X, weights, random_state=random_state)
+            else:
+                X, y, weights = shuffle(
+                    X, y, weights, random_state=random_state
+                )
+
+            if self.weighting == "mom":
+                final_weight += weights
+
+        if self.weighting == "mom":
+            self.weights_ = final_weight / self.max_iter
+        else:
+            self.weights_ = weights
+        self.base_estimator_ = base_estimator
+        self.n_iter_ = self.max_iter * len(X)
+        return self
+
+    def _get_loss_function(self, loss):
+        """Get concrete ''LossFunction'' object for str ''loss''. """
+        if type(loss) == str:
+            eff_loss = LOSS_FUNCTIONS.get(loss)
+            if eff_loss is None:
+                raise ValueError("The loss %s is not supported. " % self.loss)
+
+            loss_class, args = eff_loss[0], eff_loss[1:]
+            return np.vectorize(loss_class(*args).dloss)
```

Comment:
The attribute name here is different depending on the scikit-learn version  due to https://github.com/scikit-learn/scikit-learn/pull/17191,
```suggestion
			  # at the top of the file
			  from distutils.version import LooseVersion
			
			  if LooseVersion(sklearn.__version__) >= LooseVersion("0.24.0"):
			      dloss_attr = 'py_dloss'
			  else:
			      dloss_attr = 'dloss'
			
            return np.vectorize(geattr(loss_class(*args), dloss_attr))
```

that should fix some error in the Python 3.8 CI.